src/components/error-boundary.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false, error: null };
+  }
+
+  static getDerivedStateFromError(error) {
+    return { hasError: true, error };
+  }
+
+  componentDidCatch() {
+    // You can also log the error to an error reporting service
+    // See https://reactjs.org/docs/error-boundaries.html
+  }
+
+  render() {
+    if (this.state.hasError) {
+      const errorMessage = this.state.error ? `${this.state.error.name}: ${this.state.error.message}` : 'Unexpected error';
+      return (
+        <div className="error-boundary">
+          <div className="error-boundary-header">Oops! Something went wrong :(</div>
+          <div className="error-boundary-details">{errorMessage}</div>
+        </div>);
+    }
```

Comment:
We should tell user to contact support with a suggestion to show screenshot with error. It also means that screenshot should be useful. `error.message` is, probably, not useful enough.

Should we dump error to console and suggest user to make screenshot of console instead? I am not sure that showing stacktrace on screen can be considered user-friendly.

Reply:
The error is logged to the console as usual, but I am not sure if opening the console AFTER the error has occurred is going to reveal the stacktrace. Shouldn’t it be already open when error occurs to be logged? Anyway, I agree that advising to contact support is good, and that dumping the error to screen is not used-friendly, so I propose to keep the message it as it is now because it is an improvement, and the old way looking into the console is still there.

Reply:
<img width="707" alt="Screenshot 2019-09-09 at 21 46 45" src="https://user-images.githubusercontent.com/632081/64536283-66aafa80-d34b-11e9-96c1-24e45cb42655.png">

Does this look right?